src/ivc/cyclefold/sfc/mod.rs

```diff
@@ -78,9 +102,34 @@ where
         instances
     }
 
-    pub fn instances(&self, expected_out: CMain::ScalarExt) -> Vec<Vec<CMain::ScalarExt>> {
+    pub fn instances(
+        &self,
+        self_acc: &nifs::protogalaxy::AccumulatorInstance<CMain>,
+        support_acc: &nifs::sangria::RelaxedPlonkInstance<CSup, { support_circuit::INSTANCES_LEN }>,
+        z_out: &[CMain::ScalarExt; ARITY],
+    ) -> Vec<Vec<CMain::ScalarExt>> {
+        let _span = info_span!("consistency_marker").entered();
+
+        let mut self_ = self.input.clone();
+
+        self_.step += 1;
```

Comment:
IIUC, instances is return the instance of current step.
can you remind me that before call `fn instances`, what data the self hold/account for?

Reply:
This is the input of the current step, all possible settings. We only pass externally what will be changed by the end of the